lib/AST/Decl.cpp

```diff
@@ -2288,6 +2301,21 @@ AccessLevel ValueDecl::getFormalAccessImpl(const DeclContext *useDC) const {
   return getFormalAccess();
 }
 
+AccessLevel ValueDecl::getFormalAccess(const DeclContext *useDC,
+                                       bool treatUsableFromInlineAsPublic) const {
+  ASTContext &ctx = getASTContext();
+  AccessLevel result = ctx.evaluator(AccessLevelRequest{const_cast<ValueDecl *>(this)});
+  if (treatUsableFromInlineAsPublic &&
+      result == AccessLevel::Internal &&
+      isUsableFromInline()) {
+    return AccessLevel::Public;
+  }
+  if (useDC && (result == AccessLevel::Internal ||
+                result == AccessLevel::Public))
+    return getFormalAccessImpl(useDC);
```

Comment:
Can you pass the access level down into getFormalAccessImpl() instead of calling the getter again?

Reply:
Can do better: getFormalAccessImpl only has 3 meaningful lines and getFormalAccess is no longer an inline "fast path" (it calls r-e) so I just moved the meaningful lines from getFormalAccessImpl inline here.